### PR TITLE
Translation + Var Changes.

### PR DESCRIPTION
**Pull request by brettpenzer123**

### CODE REVIEW

**Comment by DominicMaas** [comment]
I’ll take a look at this when I get home.

**Comment by brettpenzer123** [comment]
> I’ll take a look at this when I get home.

Thanks.

File: src/Localisation/Languages/de.xml [hunk 1]
```diff
@@ -0,0 +1,106 @@
...
+  <translation id="ChatCMDnonplayers">Sie hosten keinen Server und sind mit keinem verbunden.</translation>
+  <translation id="ChatCMDhide-chat">Versteckt das Chat-Fenster. Kann durch Drücken der Taste ~ (Tilde) wieder angezeigt werden.</translation>
+  <translation id="ChatCMDclear">Löscht alles aus dem Chat-Log.</translation>
+  <translation id="ChatCMDopen-log">Öffnet den Multiplayer-Log.</translation>
+
+  <!--Multiplayer Menu-->
+  <translation id="Multiplayer">Mehrspieler</translation>
+  <translation id="MultiplayerMenu">Mehrspieler-Menü</translation>
+  <translation id="JoinGame">Spiel beitreten</translation>
+  <translation id="HostGame">Server hosten</translation>
```
Comment (hibo98) [inline src/Localisation/Languages/de.xml:26]: I think `Spiel hosten` would be a better translation.

File: src/Networking/IPAddress.cs [hunk 2]
```diff
@@ -7,7 +7,7 @@ namespace CSM.Networking
     public class IPAddress
     {
         private static string _localIp = null;
-        private static string _externalIp = null;
+        public static string _externalIp = null;
```
Comment (hibo98) [inline src/Networking/IPAddress.cs:10]: The variable should be private, only accessible with the method below, i think.

File: src/Panels/ChatLogPanel.cs [hunk 3]
```diff
@@ -44,52 +47,46 @@ public ChatLogPanel()
...
                     {
                         PrintGameMessage(player);
                     }
                 }),
-                new ChatCommand("hide-chat", "Hides the chat console. Can be displayed again by pressing the '`' key.", (command) =>
+                new ChatCommand("hide-chat", Translation.PullTranslation("ChatCMDhide-chat"), (command) =>
                 {
                     isVisible = !isVisible;
                 }),
-                new ChatCommand("donate", "Find out how to support the mod developers", (command) =>
```
Comment (hibo98) [inline src/Panels/ChatLogPanel.cs:82]: Why did you remove the `donate` command?

File: src/Panels/ConnectionPanel.cs [hunk 4]
```diff
@@ -157,7 +182,25 @@ public override void Start()
                 isVisible = false;
             };
 
-            base.Start();
```
Comment (hibo98) [inline src/Panels/ConnectionPanel.cs:160]: Superclass should also be called or why did you remove?

**Review by hibo98 — CHANGES_REQUESTED** [review]
I looked into your pull request, added some comments and found some things that got messed up.

File: src/Helpers/UiHelpers.cs [hunk 5]
```diff
@@ -113,10 +109,66 @@ public static UICheckBox CreateCheckBox(this UIComponent uiComponent, string tex
...
+            });
+
+            return dropDown;
+        }
+
+        // Sourced from : https://github.com/SamsamTS/CS-MoveIt/blob/master/MoveIt/GUI/UIUtils.cs
         // I found his code after I started this class, the below atlas feature is quite neat!
 
-        private static Dictionary<string, UITextureAtlas> _atlases;
+        private static Dictionary<string, UITextureAtlas>_atlases;
```
[the remark this answers is not in the document]
Comment (brettpenzer123) [reply src/Helpers/UiHelpers.cs:171]: Yeah the spaces need to all go back. I did try take a short cut with the translation but it went in the whole entire solution therefore when this commit was pushed I was going to rebase and start all the other commits again since I'm around 26 commits ahead of the master.

File: src/Localisation/Languages/de.xml (hunk 1 above)
Comment (brettpenzer123) [reply src/Localisation/Languages/de.xml:26]: Translation would be open to anyone finding the best context for certain text within the mod. So you're probably right.

File: src/Networking/IPAddress.cs (hunk 2 above)
Comment (brettpenzer123) [reply src/Networking/IPAddress.cs:10]: Since the server list would send the given external ip as SQL to a database it was better to change the existing one ready for another commit later on. Though it could always return to its previous one for now.

File: src/Panels/ChatLogPanel.cs (hunk 3 above)
Comment (brettpenzer123) [reply src/Panels/ChatLogPanel.cs:82]: IMO I don't think anyone would use the game command to donate for the mod in-game. Since the donation is already pushed through the steam workshop page and the github. If someone actually used it. I believe it would make sense to have it in.

File: src/Panels/ConnectionPanel.cs (hunk 4 above)
Comment (brettpenzer123) [reply src/Panels/ConnectionPanel.cs:160]: As of now I'm on my phone so I'll see the biggest changes at home but I'll close this and restart the entire thing over again. This is becoming more complicated to try separate different commits for certain pull requests. #donotmultitask
